hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALEdit.java

```diff
@@ -48,15 +48,12 @@
  * Used in HBase's transaction log (WAL) to represent a collection of edits (Cell/KeyValue objects)
  * that came in as a single transaction. All the edits for a given transaction are written out as a
  * single record, in PB format, followed (optionally) by Cells written via the WALCellEncoder.
- * <p>This class is LimitedPrivate for CPs to read-only. The {@link #add} methods are
```

Comment:
Sorry to ask you to back this out, but discussion on the JIRA ratholed on whether or not WALEdit should be immutable or not. We can keep the hook signature as proposed, because WALEdit is useful to the coprocessor even if read only, and this would allow the discussion on WALEdit to be tabled until some future time without consequence to today's need for this change.